### ISSUE TITLE
Add input to change `gha_creds_*.json` file location

### ISSUE THREAD

**Issue by mbrancato** [google-github-actions/auth#264]
### TL;DR

An input that overrides the `env.GITHUB_WORKSPACE` used as the path for storing the resulting credentials. This conflicts with checks that look for changed / uncommitted files and for use-cases where `actions/checkout` is run an additional time later in the job or in a nested composite action later.

### Detailed design

Add an input for credentials_path that overrides the use of `env.GITHUB_WORKSPACE` to be a configurable base path.

https://github.com/google-github-actions/auth/blob/23620afd0f216f749395e1c4f798d973a0cd329a/src/main.ts#L180-L184

The workspace defaults to `/home/runner/work/{{ repo }}/{{ repo }}` and that is also where code is checked out. There is some risk that the creds file gets committed to the repo if the user has an action that does `git add .` or similar.

Example config:

```yaml
jobs:
  job_id:
    # ...

    # Add "id-token" with the intended permissions.
    permissions:
      contents: 'read'
      id-token: 'write'

    steps:
    # actions/checkout MUST come before auth
    - uses: 'actions/checkout@v3'

    - id: 'auth'
      name: 'Authenticate to Google Cloud'
      uses: 'google-github-actions/auth@v1'
      with:
        credentials_path: '/home/runner/work/.config/gcloud/'
        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
```

**Comment by sethvargo** [comment]
Duplicate of https://github.com/google-github-actions/auth/issues/109

Duplicate of https://github.com/google-github-actions/auth/issues/134

Similar to https://github.com/google-github-actions/auth/issues/123

Similar to https://github.com/google-github-actions/auth/issues/212

This is also documented heavily in the README and TROUBLESHOOTING documentation.

**Comment by sethvargo** [comment]
I'm not sure I follow:

> This conflicts with checks that look for changed / uncommitted files...

This is the exact use case I described above.

> ...and for use-cases where actions/checkout is run an additional time later in the job or in a nested composite action later.

You can always run the `auth` action again, which I would definitely recommend doing in composite actions. It feels weird that your composite action would assume authentication existed (but not assume that a checkout existed). You can also use `path` on the checkout action to check out to a different path.

You're free to move or rename the creds outside of the default path, but this introduces a security risk for self-hosted runners, so we intentionally don't expose such functionality in the package.
